.github/workflows/darwin.yaml

```diff
@@ -103,40 +98,56 @@ jobs:
+                  export TEST_RUNNER_ASAN_OPTIONS=__CURRENT_VALUE__:detect_stack_use_after_return=1
 
-                  xcodebuild test -target "Matter" -scheme "Matter Framework Tests" -sdk macosx -enableThreadSanitizer YES OTHER_CFLAGS='${inherited} -Werror -Wconversion' CHIP_IS_BLE=NO GCC_PREPROCESSOR_DEFINITIONS='${inherited} MTR_NO_AVAILABILITY=1' > >(tee /tmp/darwin/framework-tests/darwin-tests-tsan.log) 2> >(tee /tmp/darwin/framework-tests/darwin-tests-tsan-err.log >&2)
-              working-directory: src/darwin/Framework
-            - name: Build Matter TV Casting Bridge
-              run: |
-                  xcodebuild -target "MatterTvCastingBridge" -sdk iphoneos
-              working-directory: examples/tv-casting-app/darwin/MatterTvCastingBridge
+                  # Disable BLE (CHIP_IS_BLE=NO) because the app does not have the permission to use it and that may crash the CI.
+                  xcodebuild test -target "Matter" -scheme "Matter Framework Tests" -sdk macosx ${{ matrix.options.arguments }} \
+                    OTHER_CFLAGS='${inherited} -Werror -Wconversion' CHIP_IS_BLE=NO GCC_PREPROCESSOR_DEFINITIONS='${inherited} MTR_NO_AVAILABILITY=1 ${{ matrix.options.defines }}' \
+                    > >(tee /tmp/darwin/framework-tests/darwin-tests.log) 2> >(tee /tmp/darwin/framework-tests/darwin-tests-err.log >&2)
             - name: Collect crash logs
               run: |
                   mkdir -p /tmp/darwin/framework-tests
                   find ~/Library/Developer/Xcode/DerivedData /Library/Logs/DiagnosticReports -name '*.ips' -print0 | xargs -0 -J % cp % /tmp/darwin/framework-tests
             - name: Uploading log files
               uses: actions/upload-artifact@v4
-              if: ${{ failure() && !env.ACT }}
+              if: failure() && !env.ACT
               with:
-                  name: darwin-framework-test-logs
+                  name: darwin-framework-test-logs-${{ matrix.options.flavor }}
                   path: /tmp/darwin/framework-tests
                   retention-days: 5
+
+    tv-casting-bridge:
+        name: Build TV Casting Bridge example
+        if: github.actor != 'restyled-io[bot]'
+        runs-on: macos-latest
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
+            - name: Setup Environment
+              run: brew install python@3.9
+            - name: Checkout submodules & Bootstrap
+              uses: ./.github/actions/checkout-submodules-and-bootstrap
+              with:
+                platform: darwin
+            - name: Build
+              working-directory: examples/tv-casting-app/darwin/MatterTvCastingBridge
+              run: xcodebuild -target "MatterTvCastingBridge" -sdk iphoneos
+
+    darwin:
+        name: Build Darwin
+        needs: [ framework, tests ]
+        runs-on: macos-latest
+        steps:
+            - name: Done
+              run: 'true' # nothing to do
```

Comment:
It would be good to have a comment here explaining why this is needed.  Just because we have an existing thing that checks that the "darwin" job succeeded, right?

Reply:
Yes, that was the idea, though I'm not sure that even works anyway

Reply:
It does seem to work. I've put in a comment, but it's really just a temporary solution until the CI requirements for PRs have been switched over to directly point to the underlying framework build and test tasks. (It'll be nicer to see those correctly listed as "Required" in the checks list rather than indirectly via the aggregation job.)